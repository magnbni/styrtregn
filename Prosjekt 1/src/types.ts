interface Day {
    day: Record<string, JSONArray>
}

interface Geo {
    lat: string,
    lon: string
}

interface WeatherData {
    date: string
}

interface Hour {
    temp_c: string,
    time: string,
    condition: Condition
}

interface Condition {
    icon: string
}

type JSONvalue = string | number | string[] | Array<JSONvalue> | Hour;



interface JSONArray extends Array<Hour> { }