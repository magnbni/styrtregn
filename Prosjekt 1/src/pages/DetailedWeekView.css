.detcomp {
    display: flex;
    flex-direction: column;
    position: relative;
    border-radius: 10px;
    width: 96%;
    overflow-y: auto;
    padding-left: 2%;
    padding-right: 2%;
}

.detrow {
    display: flex;
    flex-direction: row;
    justify-content: space-between; /* Space between cells */
    border-bottom: 2px solid black;
}

.detcell {
    display: flex;
    align-items: center;
    justify-content: center;
    flex: 1; /* To distribute cell space evenly */
    margin: 0 2%; /* Add some margin for space between cells */
    color: white;
}


.weathericon {
    width: 90%;
}
