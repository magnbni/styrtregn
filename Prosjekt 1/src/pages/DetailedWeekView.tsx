import "./DetailedWeekView.css";

export default function DetWeekView() {
  

  return (
    <div className="detcomp">
      <div className="detrow">
        <p className="detcell">Dato</p>
        <p className="detcell">Natt</p>
        <p className="detcell">Morgen</p>
        <p className="detcell">Ettermiddag</p>
        <p className="detcell">Kveld</p>
        <p className="detcell">Maks/min. temp.</p>
        <p className="detcell">Nedb√∏r</p>
        <p className="detcell">Vind</p>
      </div>
      <div className="detrow">
        <p className="detcell">Detaljert</p>
        <p className="detcell">Detaljert</p>
      </div>
    </div>
  );
}
