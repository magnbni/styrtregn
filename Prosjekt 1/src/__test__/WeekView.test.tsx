import { describe, expect, it } from 'vitest'
import DetailedWeekView from '../components/WeekView';
import { DayView, Properties } from '../types';
import { defaultMetData } from '../test/constants';
import { act, render } from '@testing-library/react';

describe('DetailedWeekView test', ()=>{
    const props: {dayView: DayView; rawData: Properties} = {
        dayView: [{
            date: "2023-06-14",
            rain: 5,
            wind: 3,
            maxTemp: 15,
            symbol: "noncloud"
        }],
        rawData: defaultMetData.properties
    }

    const detailedWeekView = render(<DetailedWeekView dayView={props.dayView} rawData={props.rawData}/>)
    
    it('The DetailedWeekView component should output the same HTML with the same props', () => {
        expect(detailedWeekView).toBeDefined()
        expect(detailedWeekView).toMatchSnapshot()
    })
    
    it("The HourlyView should exist when button is pressed", () => {
        const nowWithButton = render(<DetailedWeekView dayView={props.dayView} rawData={props.rawData}></DetailedWeekView>)
        expect(nowWithButton).toBeDefined()
        const hourlyViewButton = detailedWeekView.getByPlaceholderText("hourView")
        expect(hourlyViewButton).toBeTruthy()
        act(()=>{hourlyViewButton.click()})
        if(props.dayView){
            // Testing if the popup header text now exists
            const isShowing = nowWithButton.getByText(`VÃ¦ret ${props.dayView[0].date}`)
            expect(isShowing).toBeTruthy()
        } else {
            expect(false).toBeTruthy()
        }
    })
})