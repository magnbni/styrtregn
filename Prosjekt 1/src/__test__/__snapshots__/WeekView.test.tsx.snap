// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DetailedWeekView test > The DetailedWeekView component should output the same HTML with the same props 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="detcomp"
      >
        <div
          class="detrow"
        >
          <p
            class="detcell"
          >
            Dato
          </p>
          <p
            class="detcell"
          >
            Vær
          </p>
          <p
            class="detcell"
          >
            Min/Max. temp.
          </p>
          <p
            class="detcell"
          >
            Nedbør
          </p>
          <p
            class="detcell"
          >
            Vind
          </p>
          <p
            class="detcell"
          >
            Oversikt
          </p>
        </div>
        <div
          class="detrow"
        >
          <p
            class="detcell"
          >
            14
            .
            06
             
            Onsdag
          </p>
          <div
            class="detcell"
          >
            <img
              alt="icon"
              class="weathericon"
            />
          </div>
          <p
            class="detcell"
          >
            15
             
            °
            C
          </p>
          <p
            class="detcell"
          >
            5
             mm
          </p>
          <p
            class="detcell"
          >
            3
             m/s
          </p>
          <p
            class="detcell"
          >
            <button
              placeholder="hourView"
            >
              Se været
            </button>
          </p>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="detcomp"
    >
      <div
        class="detrow"
      >
        <p
          class="detcell"
        >
          Dato
        </p>
        <p
          class="detcell"
        >
          Vær
        </p>
        <p
          class="detcell"
        >
          Min/Max. temp.
        </p>
        <p
          class="detcell"
        >
          Nedbør
        </p>
        <p
          class="detcell"
        >
          Vind
        </p>
        <p
          class="detcell"
        >
          Oversikt
        </p>
      </div>
      <div
        class="detrow"
      >
        <p
          class="detcell"
        >
          14
          .
          06
           
          Onsdag
        </p>
        <div
          class="detcell"
        >
          <img
            alt="icon"
            class="weathericon"
          />
        </div>
        <p
          class="detcell"
        >
          15
           
          °
          C
        </p>
        <p
          class="detcell"
        >
          5
           mm
        </p>
        <p
          class="detcell"
        >
          3
           m/s
        </p>
        <p
          class="detcell"
        >
          <button
            placeholder="hourView"
          >
            Se været
          </button>
        </p>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
